arrows/kpf/yaml/kpf_yaml_parser.cxx

```diff
@@ -188,51 +188,59 @@ bool
 parse_packet( const YAML::const_iterator& it, KPF::packet_t& p )
 {
   bool okay = true;
-  switch (p.header.style)
+  try
   {
-  case KPF::packet_style::TS:
-    new (&p.timestamp) KPFC::timestamp_t( it->second.as<double>() );
-    break;
+    switch (p.header.style)
+    {
+    case KPF::packet_style::TS:
+      new (&p.timestamp) KPFC::timestamp_t( it->second.as<double>() );
+      break;
 
-  case KPF::packet_style::CONF:
-    new (&p.conf) KPFC::conf_t( it->second.as<double>() );
-    break;
+    case KPF::packet_style::CONF:
+      new (&p.conf) KPFC::conf_t( it->second.as<double>() );
+      break;
 
-  case KPF::packet_style::CSET:
-    okay = parse_cset( p, it->second );
-    break;
+    case KPF::packet_style::CSET:
+      okay = parse_cset( p, it->second );
+      break;
 
-  case KPF::packet_style::EVAL:
-    new (&p.eval) KPFC::eval_t( it->second.as<double>() );
-    break;
+    case KPF::packet_style::EVAL:
+      new (&p.eval) KPFC::eval_t( it->second.as<double>() );
+      break;
 
-  case KPF::packet_style::ID:
-    new (&p.id) KPFC::id_t( it->second.as<int>() );
-    break;
+    case KPF::packet_style::ID:
+      new (&p.id) KPFC::id_t( it->second.as<int>() );
+      break;
 
-  case KPF::packet_style::KV:
-    new (&p.kv) KPFC::kv_t( it->first.as<string>(), it->second.as<string>() );
-    break;
+    case KPF::packet_style::KV:
+      new (&p.kv) KPFC::kv_t( it->first.as<string>(), it->second.as<string>() );
+      break;
 
-  case KPF::packet_style::META:
-    new (&p.meta) KPFC::meta_t( it->second.as<string>() );
-    p.header.domain = KPF::packet_header_t::NO_DOMAIN;
-    break;
+    case KPF::packet_style::META:
+      new (&p.meta) KPFC::meta_t( it->second.as<string>() );
+      p.header.domain = KPF::packet_header_t::NO_DOMAIN;
+      break;
 
-  case KPF::packet_style::POLY:
-    okay = parse_poly( p, it->second );
-    break;
+    case KPF::packet_style::POLY:
+      okay = parse_poly( p, it->second );
+      break;
 
-  case KPF::packet_style::GEOM:
-    okay = parse_geom( p, it->second.as<string>() );
-    break;
+    case KPF::packet_style::GEOM:
+      okay = parse_geom( p, it->second.as<string>() );
+      break;
 
-  default:
-    {
-      LOG_ERROR( main_logger, "No implementation for parsing packet of type " << p.header );
-      return false;
-    }
-  } // ...switch
+    default:
+      {
+        LOG_ERROR( main_logger, "No implementation for parsing packet of type " << p.header );
+        okay = false;
+      }
+    } // ... switch
+  } // ... try
+  catch (const YAML::Exception& e )
```

Comment:
Begs the questions, is a failure fatal and if so, should we throw? Not sure what the current kwiver protocol is regarding exception throwing.

Reply:
Currently the rest of the code uses return codes rather than exceptions; this basically converts the YAML-cpp library's exception into a return code.

Regarding whether or not a parse error is fatal... I was wondering about that. In general, and in KPF in particular, I think it's good when the code can ignore **semantically unknown** items (Postel's Law). The KPF reader converts each line into a bucket of packets, and it's up to the client to pull what it's looking for and ignore the rest (via `reader.flush()`).

But I think **ill-formed** packets are a sign of a more fundamental (error, issue, misunderstanding) between the writer and the reader.

Reply:
I tend to throw exceptions when the error should not be ignored, since error return codes are frequently ignored. Unless it causes a performance problem. I would vote for ill-formed packets to throw and semantically unknown packets to return an error.

Reply:
The upstream caller checks this return code; I'm happy to switch to exceptions but would prefer to do that as part of a larger overhaul on a separate branch.

Reply:
I have no problem with pushing off any exceptional changes to a future date.

Reply:
Does that mean we're ready to merge this sucker?